mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td

```diff
@@ -538,6 +587,22 @@ def NVVM_MBarrierInvalSharedOp : NVVM_Op<"mbarrier.inval.shared">,
 def NVVM_MBarrierArriveOp : NVVM_Op<"mbarrier.arrive">,
   Results<(outs LLVM_Type:$res)>,
   Arguments<(ins LLVM_AnyPointer:$addr)> {
+  let summary = "MBarrier Arrive Operation";
+  let description = [{
+    The `nvvm.mbarrier.arrive` operation performs an arrive-on operation on the 
+    mbarrier object at the specified address. Uses the default `.release.cta` semantics.
+
+    This operation causes the executing thread to signal its arrival at the barrier.
+    The operation returns an opaque 64-bit value that captures the phase of the 
```

Comment:
Specifying bitsize in the dialect isn't maintainable. It's easier to have them as Types, so they can automatically rendered and type checked. 

Can we change the result?
```
 Results<(outs LLVM_Type:$res)>
```
to 
```
 Results<(outs I64:$res)>
```